src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityKMSRequest.java

```diff
@@ -187,4 +190,29 @@ private TenantSecurityKMSException parseFailureFromRequest(HttpResponse resp) {
             this.batchWrapEndpoint);
         return this.makeRequestAndParseFailure(this.batchUnwrapEndpoint, postData, BatchUnwrappedDocumentKeys.class, error);
     }
+
+    /**
+     * Create the factory for http requests.
+     * The main reason for this is to provide connection pooling for situations where large numbers of
+     * requests are desired.
+     *
+     * @param maxConnections        global max connections
+     * @param maxRouteConnections   max connections for a single HTTP endpoint
+     * @return HttpRequestFactory with connection pooling enabled.
+     */
+    static HttpRequestFactory provideHttpRequestFactory(int maxConnections, int maxRouteConnections) {
+        final PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager();
+        // Increase max total connections
+        cm.setMaxTotal(maxConnections);
+        // Increase default max connection per route
+        cm.setDefaultMaxPerRoute(maxRouteConnections);
+
+        final CloseableHttpClient httpClient = HttpClients
+                .createMinimal(cm);
+        final HttpTransport httpTransport = new ApacheHttpTransport(httpClient);
+
+        return httpTransport.createRequestFactory((HttpRequest request) -> {
+            request.setParser(new JsonObjectParser(JSON_FACTORY));
```

Comment:
Is this use of `JSON_FACTORY` redundant with the one in `getApiRequest`?

Reply:
I don't believe it is. One sets the ability to use that parser and the other tells it to use that parser. When he didn't have this line yesterday he was getting a runtime exception.